lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -1888,3 +1979,56 @@ LargeHeapBlock::GetTrackerDataArray()
     return (void **)((char *)(this + 1) + LargeHeapBlock::GetAllocPlusSize(this->objectCount) - this->objectCount * sizeof(void *));
 }
 #endif
+
+#ifdef RECYCLER_PAGE_HEAP
+void
+LargeHeapBlock::CapturePageHeapAllocStack()
+{
+    if (this->InPageHeapMode()) // pageheap can be enabled only for some of the buckets
+    {
+        // These asserts are true because explicit free is disallowed in
+        // page heap mode. If they weren't, we'd have to modify the asserts
+        Assert(this->pageHeapFreeStack == nullptr);
+        Assert(this->pageHeapAllocStack == nullptr);
+
+        // Note: NoCheckHeapAllocator will fail fast if we can't allocate the stack to capture
+        // REVIEW: Should we have a flag to configure the number of frames captured?
+        if (pageHeapAllocStack != nullptr && this->pageHeapAllocStack != (StackBackTrace*)1)
+        {
+            this->pageHeapAllocStack->Capture(Recycler::s_numFramesToSkipForPageHeapAlloc);
+        }
+        else
+        {
+            this->pageHeapAllocStack = StackBackTrace::Capture(&NoThrowHeapAllocator::Instance,
+                Recycler::s_numFramesToSkipForPageHeapAlloc, Recycler::s_numFramesToCaptureForPageHeap);
+        }
+        
+        if (this->pageHeapAllocStack == nullptr) 
+        {
+            this->pageHeapAllocStack = (StackBackTrace*)1; // allocate failed, mark it we have tried
```

Comment:
Ah, I see- consider changing this to a sentinel constant rather than using magic numbers